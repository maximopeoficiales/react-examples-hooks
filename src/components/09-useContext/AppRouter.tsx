import React from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  NavLink,
} from "react-router-dom";
import AboutPage from "./AboutPage/AboutPage";
import HomePage from "./HomePage/HomePage";
import LoginPage from "./LoginPage/LoginPage";

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <nav className="navbar navbar-expand navbar-light bg-light">
          <div className="nav navbar-nav">
            <NavLink
              activeClassName="active"
              exact
              className="nav-item nav-link "
              to="/"
            >
              Home
            </NavLink>
            <NavLink
              activeClassName="active"
              exact
              to="/about"
              className="nav-item nav-link"
            >
              About
            </NavLink>
            <NavLink
              activeClassName="active"
              exact
              to="/login"
              className="nav-item nav-link"
            >
              Login
            </NavLink>
          </div>
        </nav>
        <div className="my-4">
          <Switch>
            <Route exact path="/" component={HomePage} />
            <Route exact path="/about" component={AboutPage} />
            <Route exact path="/login" component={LoginPage} />
          </Switch>
        </div>
      </div>
    </Router>
  );
};
