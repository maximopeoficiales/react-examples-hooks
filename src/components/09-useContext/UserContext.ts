import React, { createContext } from 'react'
export interface User {
    id?: number,
    name?: string,
    age?: number,
}
interface UserProvider {
    user?: User,
    setUser?: any
}

export const UserContext = createContext<UserProvider>({});