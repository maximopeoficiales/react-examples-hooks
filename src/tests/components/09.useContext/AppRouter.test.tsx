import { mount } from "enzyme";
import React from "react";
import { AppRouter } from "../../../components/09-useContext/AppRouter";
import {
  User,
  UserContext,
} from "../../../components/09-useContext/UserContext";

describe("Pruebas en LoginPage", () => {
  // mount monta el componente con todos sus hijos
  const user: User = {
    id: 1,
    age: 33,
    name: "jorge",
  };

  const wrapper = mount(
    <UserContext.Provider value={{ user, setUser: () => {} }}>
      <AppRouter />
    </UserContext.Provider>
  );
  test("debe de mostrarse correctament", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
