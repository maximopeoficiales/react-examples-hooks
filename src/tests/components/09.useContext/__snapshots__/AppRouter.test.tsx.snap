// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginPage debe de mostrarse correctament 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-light bg-light"
  >
    <div
      className="nav navbar-nav"
    >
      <a
        aria-current="page"
        className="nav-item nav-link  active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </nav>
  <div
    className="my-4"
  >
    <div
      className="row justify-content-center"
      data-testid="HomePage"
    >
      <div
        className="col-md-10"
      >
        <h1>
          HomePage component
        </h1>
        <pre>
          {
   "id": 1,
   "age": 33,
   "name": "jorge"
}
        </pre>
      </div>
    </div>
  </div>
</div>
`;
